google-cloud-graalvm-samples/graalvm-samples-client-library/bigtable-sample/src/main/java/com/example/BigTableSampleApplication.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import com.google.api.gax.rpc.ServerStream;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowCell;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * Sample Cloud BigTable application.
+ */
+public class BigTableSampleApplication {
+
+  private static final String INSTANCE_NAME = "graalvm-test-instance";
```

Comment:
Instead of hardcoding this maybe use `System.getProperty(name, default)`, and your instructions can be a little looser? 

It's a bit odd to me, also, to require the instance to exist but autogenerate the table - could you autocreate both? Or neither?

Reply:
Done, switched to a property.

Yeah I agree that it is a bit awkward to ask the user to have an instance ready but then auto-create the table. However I decided to do this because creating a BigTable instance is a long-running/flaky operation, and creating instances is not supported on the BigTable emulator. Hopefully since this is all just toy code it won't matter too much.